tests/Functional/Parameters/PartialSearchFilterTest.php

```diff
@@ -0,0 +1,164 @@
+use PHPUnit\Framework\Attributes\DataProvider;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+
+/**
+ * @author Vincent Amstoutz <vincent.amstoutz.dev@gmail.com>
+ */
+final class PartialSearchFilterTest extends ApiTestCase
+{
+    use RecreateSchemaTrait;
+    use SetupClassResourcesTrait;
+
+    protected static ?bool $alwaysBootKernel = false;
+
+    /**
+     * @return class-string[]
+     */
+    public static function getResources(): array
+    {
+        return [Chicken::class, ChickenCoop::class];
+    }
+
+    /**
+     * @throws \Throwable
+     */
+    protected function setUp(): void
+    {
+        $entities = $this->isMongoDB()
+            ? [DocumentChicken::class, DocumentChickenCoop::class]
+            : [Chicken::class, ChickenCoop::class];
+
+        $this->recreateSchema($entities);
+        $this->loadFixtures();
+    }
+
+    /**
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     * @throws TransportExceptionInterface
+     */
+    #[DataProvider('partialSearchFilterProvider')]
+    public function testPartialSearchFilter(string $url, int $expectedCount, array $expectedNames): void
+    {
+        $response = self::createClient()->request('GET', $url);
+        $this->assertResponseIsSuccessful();
+
+        $responseData = $response->toArray();
+        $filteredItems = $responseData['hydra:member'];
+
+        $this->assertCount($expectedCount, $filteredItems, \sprintf('Expected %d items for URL %s', $expectedCount, $url));
+
+        $names = array_map(fn ($chicken) => $chicken['name'], $filteredItems);
+        sort($names);
+        sort($expectedNames);
+
+        $this->assertSame($expectedNames, $names, 'The returned names do not match the expected values.');
+    }
+
+    public static function partialSearchFilterProvider(): \Generator
+    {
+        yield 'filter by partial name "ertrude"' => [
+            '/chickens?namePartial=ertrude',
+            1,
+            ['Gertrude'],
+        ];
+
+        yield 'filter by partial name "riette"' => [
+            '/chickens?namePartial=riette',
+            1,
+            ['Henriette'],
+        ];
+
+        yield 'filter by partial name "e" (should match both)' => [
+            '/chickens?namePartial=e',
+            2,
+            ['Gertrude', 'Henriette'],
+        ];
+
+        yield 'filter by partial name with no matching entities' => [
+            '/chickens?namePartial=Zebra',
+            0,
+            [],
+        ];
+
+        yield 'filter with multiple partial names "rude" OR "iette"' => [
```

Comment:
this is an AND I think

Reply:
Oh indeed it's querying `IN (rude, iette)`

Reply:
Yes, so it's ok no? @soyuka